src/context/AppProvider.jsx

```diff
@@ -225,14 +241,21 @@ export const AppProvider = ({ children }) => {
   useInterval(
     async () => {
       if (coinContracts == null || !isVisible) return;
-      const coinsDetails = await getCoinDetails(
-        coinContracts.stableCoin,
-        coinContracts.reserveCoin,
-        djedContract,
-        decimals.scDecimals,
-        decimals.rcDecimals,
-        oracleContract
-      );
+      const coinsDetails = isShu
+        ? await getShuCoinDetails(
```

Comment:
Djed and DjedShu are identical in relation to coin details. Do we really need this `isShu ?` if statement here?

Reply:
This comment has not been addressed yet.

Reply:
Yes, `isShu` condition fetches coin details from  `getShuCoinDetails` instead of  `getCoinDetails` and there are considerable differences between the two